### ISSUE TITLE
METS paths have encoding issues

### ISSUE THREAD

**Issue by hsilva-keep** [keeps/commons-ip#27]
Some time ago url encode/decode was introduced in commons-ip with the following message "Alpha22: Fixed issue related to anyURI METS fields (added URLEncode / Decode to setters and getters)" (https://github.com/keeps/commons-ip/commit/fdeacc1fc1ae238af95cdb9ca2c28ba3ea5228a7). 
The big problem is we don't know for sure what was the reason to do so, and for RODA-in -> RODA relationship it doesn't matter because build and parse is done by roda-commons-ip. For XML Schema, ASCII & non ASCII are allowed with some exceptions:
``URIs require that some characters be escaped with their hexadecimal Unicode code point preceded by the % character. This includes non-ASCII characters and some ASCII characters, namely control characters, spaces, and the following characters (unless they are used as deliimiters in the URI): <>#%{}|\^`. For example, ../édition.html must be represented instead as ../%C3%A9dition.html, with the é escaped as %C3%A9. However, the anyURI type will accept these characters either escaped or unescaped. With the exception of the characters % and #, it will assume that unescaped characters are intended to be escaped when used in an actual URI, although the schema processor will do nothing to alter them. It is valid for an anyURI value to contain a space, but this practice is strongly discouraged. Spaces should instead be escaped using %20.``

**Comment by hsilva-keep** [comment]
@luis100 wrote the following about this matter:
```txt
I am sure we need to at least test the system with characters in the URL that are illegal for the URL and for the XML.

https://perishablepress.com/stop-using-unsafe-characters-in-urls/

The table on the link above shows that the following would need to be encoded:
- ASCII Control characters: Includes the ISO-8859-1 (ISO-Latin) character ranges 00-1F hex (0-31 decimal) and 7F (127 decimal.)
- Non-ASCII characters: Includes the entire “top half” of the ISO-Latin set 80-FF hex (128-255 decimal.)
- Reserved characters: `; / ? : @ = &` (does not include blank space)
- Unsafe characters: Includes the blank/empty space and ``" < > # % { } | \ ^ ~ [ ] ` ``

@rui-castro I suggest you test the generation of METS using files with the characters above on the name and check if it is still a valid XML and URL (using xmllint and the METS and xlink schemas).
```

**Comment by hsilva-keep** [comment]
@luis100 also wrote the following:

I've improved current tests to include testing the encoding at my branch (lf_dev) after merging with your pull request:

https://github.com/keeps/commons-ip/blob/lf_dev/src/test/java/org/roda_project/commons_ip/model/impl/eark/EARKTest.java#L121

The tests **fail** because the produced METS it is **NOT valid**.

```
org.xml.sax.SAXParseException; systemId: file:/tmp/temp5635430356598914484/representations/representation%201/METS.xml; lineNumber: 13; columnNumber: 85; An invalid XML character (Unicode: 0x1) was found in the value of attribute "xlink:href" and element is "FLocat".
```

**Comment by hsilva-keep** [comment]
Solved since alpha 31 (20848122208826db7ac102899ace798806c2e6a2 & d5e1008).
